### ISSUE TITLE
Questions regarding shapr and papers

### ISSUE THREAD

**Issue by fredriklaa** [NorskRegnesentral/shapr#443]
Hi,

I have some questions regarding your software and papers:

(1) Do you perform some transformations/scaling on the observations or to the predictions when one wants to explain a logistic regression? How should phi0 be chosen?
(2) In two of your papers, you say that kernel shap can handle categorical features by using one hot encoding. Denote the prediction function by f(x) and the dependent variables by x. Are you assuming that f(x) given x is gaussian, where you apply dummy variable encoding? Is is unclear to me where the SAMPLING occurs when you have dummy variables. 

Thanks in advance

Fredrik

**Comment by fredriklaa** [comment]
Do you model log odds when you apply the gaussian model, i.e for the one-hot encoded features?

**Comment by martinju** [comment]
Hi Fredrik,

Generally, I would recommend looking into the main vignette: https://norskregnesentral.github.io/shapr/articles/general_usage.html for details about the functionality of the package. 
Brief answers to your specific questions below:

1) We don't do any type of transformation. The package is fully model-agnostic so there is no difference between logistic regression and real regression. You just define your f(x), and work with that. phi0 should be an estimate of E[f(x)]. I typically use the mean of the response, but using the mean of the predictions on a training or test set is also OK.
2) For categorical features I would recommend using ctree. One-hot-encoding combined with gaussian features is not something I would recommend.

**Comment by fredriklaa** [comment]
Thank for your quick reply. 
1) Yes, that was what I was wondering about, either mean or proportion of the response or mean prediction.
2) I would like to replicate your study, so the details of the implementations would be great to know/have.

**Comment by martinju** [comment]
I don't think we have worked with one-hot-encoding and gaussian features before. Which study are you referring to?

**Comment by fredriklaa** [comment]
**Explaining predictive models with mixed
features using Shapley values and conditional
inference trees**

Excerpt:
The empirical and Gaussian methods cannot handle categorical features. For
these methods, we transform the categorical features into one-hot encoded features. If the categorical feature originally has L categories, the one-hot encoded
transformation creates L−1 binary features representing the second, third (etc)
categories. The first category is represented by the intercept. The Shapley value
of each categorical feature is then the sum of the Shapley values of the corresponding one-hot encoded feature

**Comment by martinju** [comment]
Right, I had forgotten about that one. As you see from the results, doing one-hot-encoding + gaussian is not a good idea here.

As for your question, we when apply the gaussian approach to one-hot-encoded features, we essentially assume all the one-hot-encoded features are gaussian and carry on as in the standard case from there. The assumption is obviously completely wrong, but may work out alright in some cases depending on the influence of the features, the underlying model etc. It's hard to know when it works and not in advance, though, and the performance is better with e.g. ctree, which is why I highly recommend that instead.

**Comment by fredriklaa** [comment]
Thanks! But e.g gender is a dummy variable. How do you pass a numeric value to a function that takes female or male as input?

**Comment by martinju** [comment]
Sorry for the late reply. There are different ways of doing this, and none of them might be very good, hence my recommendation of NOT doing this. 
If you still want to do it, you can still technically do it. E.g. for a linear model coded with dummy features, you can e.g. receive half the regression coefficient for male if x=.5. You can technically do the same with e.g. xgboost which takes binary versions as input -- the splitting may be somewhat arbitrary in that case, but the point is that you can do it technically. 

Once again, not recommended :-)
